### PR TITLE
TTS recipes for thchs30 and aishell3

### PR DESCRIPTION
**Pull request by ftshijt**
This PR serves for tts recipes of thchs30 and aishell3.

There are two main issues for now:

- The thchs30 only has audio with 16,000 sr. I think most tts systems expect >22050 sr, does it OK with that sr?
- both recipes have the direct support of `phn` transcription, do we have some examples of using its built-in g2p?

Another question:
How the correctness of g2p will help the system? There are some recent generations of Mandarin g2p, if that makes good improvements, I can try to add them to the system.

### CODE REVIEW

File: egs2/aishell3/tts1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+stage=1
+stop_stage=2
+
+
+
+log "$0 $*"
+. utils/parse_options.sh
+
+
```
Comment (kan-bayashi) [inline egs2/aishell3/tts1/local/data.sh:21]:
```suggestion

stage=1
stop_stage=2

log "$0 $*"
. utils/parse_options.sh

```

File: egs2/aishell3/tts1/conf/tuning/finetune_tacotron2.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# This configuration is for ESPnet2 to finetune Tacotron 2. Compared to the
```
Comment (kan-bayashi) [inline egs2/aishell3/tts1/conf/tuning/finetune_tacotron2.yaml:1]: This configuration is for fine-tuning of single speaker case.
If the recipe is multi-speaker, please replace the config with the following one.
3d83127/egs2/vctk/tts1/conf/tuning
maybe only using the following configs is OK: (default is tacotron2 or transformer)

- train_gst+xvector_conformer_fastspeech2.yaml
- train_gst+xvector_tacotron2.yaml
- train_gst+xvector_transformer.yaml

File: egs2/aishell3/tts1/conf/finetune.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# This configuration is for ESPnet2 to finetune Tacotron 2. Compared to the
```
Comment (kan-bayashi) [inline egs2/aishell3/tts1/conf/finetune.yaml:1]: Please remove this one and create `conf/train.yaml` which is symlink to `tuning/train_gst+xvector_tacotron2.yaml`.

File: egs2/aishell3/tts1/conf/decode.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# This configuration is the basic decoding setting for Tacotron 2.
```
Comment (kan-bayashi) [inline egs2/aishell3/tts1/conf/decode.yaml:1]: Please change this one to symlink to `tuning/decode_tacotron2.yaml`.

File: egs2/thchs30/tts1/local/thchs-30_data_prep.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
```
Comment (kan-bayashi) [inline egs2/thchs30/tts1/local/thchs-30_data_prep.sh:1]: If phoneme is provided, why don't you make two data directories?
- train/dev/test: standard text
- train_phn/dev_phn/test_phn: phoneme_text

Then, we can select in `run.sh`.

File: egs2/thchs30/tts1/run.sh [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+valid_set=dev
+test_sets="dev test"
+
+train_config=conf/finetune.yaml
+inference_config=conf/decode.yaml
+
+g2p=pypinyin_g2p_phone
+# Input: 卡尔普陪外孙玩滑梯
+# pypinyin_g2p: ka3 er3 pu3 pei2 wai4 sun1 wan2 hua2 ti1
+# pypinyin_g2p_phone: k a3 er3 p u3 p ei2 uai4 s un1 uan2 h ua2 t i1
```
Comment (kan-bayashi) [inline egs2/thchs30/tts1/run.sh:31]:
```suggestion
train_config=conf/train.yaml
inference_config=conf/decode.yaml

# if you want to use on-the-fly g2p (better for online demo)
train_set=train
valid_set=dev
test_sets="dev test"
g2p=pypinyin_g2p_phone

# if you want to use officially provided phoneme text (better for the quality)
# train_set=train_phn
# valid_set=dev_phn
# test_sets="dev_phn test_phn"
# g2p=none
```

**Comment by kan-bayashi** [comment]
Thank you for making new recipes!

> The thchs30 only has audio with 16,000 sr. I think most tts systems expect >22050 sr, does it OK with that sr?

16khz is fine with different shift size or fft point.
(I added suggestion)

> both recipes have the direct support of phn transcription, do we have some examples of using its built-in g2p?

I added suggestion for thchs30. Please check it.

> How the correctness of g2p will help the system? There are some recent generations of Mandarin g2p, if that makes good improvements, I can try to add them to the system.

Actually, I'm not familiar with Mandarin so I'm not sure how it correct.
Adding new g2p is great idea! You can refer the following PR.
https://github.com/espnet/espnet/pull/3382

**Comment by ftshijt** [comment]
Thanks for the informative comments!

> Adding new g2p is great idea! You can refer the following PR.
> #3382

Cool! I will first make the recipe available with our current g2p and start a new PR to the new methods.

File: egs2/thchs30/tts1/local/data.sh [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "stage -1: download data from openslr"
+    local/download_and_untar.sh "${db_root}" "https://www.openslr.org/resources/18/data_thchs30.tgz" data_thchs30.tgz 
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "stage 1: prepare thchs30 data"
+    local/thchs-30_data_prep.sh $(pwd) "${db_root}"/data_thchs30 || exit 1;
+fi
```
[the remark this answers is not in the document]
Comment (ftshijt) [reply egs2/thchs30/tts1/local/data.sh:45]: Seems that `pyscripts/audio/trim_silence.py` is not presented in the repo. Do you have any ideas about it?

[the remark this answers is not in the document]
Comment (kan-bayashi) [reply egs2/thchs30/tts1/local/data.sh:45]: Sorry it was fixed in https://github.com/espnet/espnet/pull/3467

[the remark this answers is not in the document]
Comment (ftshijt) [reply egs2/thchs30/tts1/local/data.sh:45]: I see. Many thanks!

**Pull request merged**